api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java

```diff
@@ -3542,4 +3542,143 @@ public CohortDefinition getPatientsOnMICat18Numerator() {
+                    + "               ON e.encounter_id = o.encounter_id "
+                    + "WHERE  e.encounter_type = ${6} "
+                    + "       AND o.concept_id = ${1268} "
+                    + "       AND o.value_coded = ${1256} "
+                    + "       AND p.voided = 0 "
+                    + "       AND e.voided = 0 "
+                    + "       AND o.voided = 0 "
+                    + "       AND e.location_id = :location "
+                    + "       AND o.obs_datetime BETWEEN Date_sub(:endDate, interval 7 month) AND :endDate "
+                    + "UNION "
+                    + "SELECT p.patient_id "
+                    + "FROM   patient p "
+                    + "       inner join encounter e "
+                    + "               ON e.patient_id = p.patient_id "
+                    + "       inner join obs o "
+                    + "               ON o.encounter_id = e.encounter_id "
+                    + "WHERE  e.encounter_type = ${6} "
+                    + "       AND o.concept_id = ${23761} "
+                    + "       AND o.value_coded = ${1065} "
+                    + "       AND p.voided = 0 "
+                    + "       AND e.voided = 0 "
+                    + "       AND o.voided = 0 "
+                    + "       AND e.location_id = :location "
+                    + "       AND e.encounter_datetime BETWEEN Date_sub(:endDate, interval 7 month) AND :endDate "
+                    + "GROUP  BY patient_id ";
+
+    StringSubstitutor stringSubstitutor = new StringSubstitutor(map);
+
+    sql.setQuery(stringSubstitutor.replace(query));
+
+
+    return sql;
+  }
+
+  public CohortDefinition getPatientsWhoReactedToTheMedication(){
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Patients Who Had any Reaction To The Medication ");
+    cd.addParameter(new Parameter("endDate", "endDate", Date.class));
+    cd.addParameter(new Parameter("location", "location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+
+
+    String query =
+            " SELECT p.patient_id "
+                    + "FROM   patient p "
+                    + "       inner join encounter e "
+                    + "               ON e.patient_id = p.patient_id "
+                    + "       inner join obs o "
+                    + "               ON o.encounter_id = e.encounter_id "
+                    + "WHERE  e.encounter_type = ${6} "
+                    + "       AND ( o.concept_id = 2015 "
+                    + "             AND o.value_coded IN ( 23748, 6293, 23749, 29, "
```

Comment:
23748, 6293, 23749, 29 and so on are not mapped, please add the corresponding mapping